subsys/logging/log_backend_adsp.c

```diff
@@ -76,7 +81,11 @@ static void process(const struct log_backend *const backend,
 {
 	log_format_func_t log_output_func = log_format_func_t_get(log_format_current);
 
+	k_spinlock_key_t key = k_spin_lock(&lock);
```

Comment:
have you considered to put this lock in `char_out` around `intel_adsp_trace_out`. Here it will lock for significant amount of time in the lower priority thread (in case of deferred mode). Putting it in `char_out` may lead to some interleaving of data within the log message so there is a trade off.

Reply:
That was one of the problems being solved here, as on the intel_adsp with a fatal exception ISR being run on one cpu, the lower priority logger thread continues to run and trample the data.

I as debating if a lock should be placed in msg_process in log_core.c instead as it seems there's a race when panic mode and the log thread collide on SMP systems.